Project2015To2017/Analysis/DiagnosticSet.cs

```diff
@@ -0,0 +1,22 @@
+using System.Collections.Generic;
+using Project2015To2017.Analysis.Diagnostics;
+
+namespace Project2015To2017.Analysis
+{
+	public class DiagnosticSet : HashSet<DiagnosticBase>
```

Comment:
Mutable until assignment to AnalysisOptions. This way the calling side can pick and remove any diagnostic before applying.